design/tracing-design.md

```diff
@@ -0,0 +1,170 @@
+- **Trace formats other than OpenTelemetry.** While Contour *may* choose to add direct support for other trace formats in the future, our hope is that by supporting the OpenTelemetry collector, users are able to export traces to the backend of their choice without having to directly configure it in Contour/Envoy.
+
+## High-Level Design
+The Contour configuration file will be extended with a new optional `tracing` section. This configuration block, if present, will enable tracing and will define the trace format and other properties needed to generate and export trace data.
+
+The `OpenTelemetry` trace format will be supported.
+
+The parameters defined in the configuration file will be used to configure each [HTTP connection manager's `tracing` section][3].
+
+Since the tracing backend is pluggable, Contour will not package any particular backend.
+The user is responsible for deploying and operating the tracing backend of their choice, and configuring Contour to make use of it.
+
+## Detailed Design
+
+### otel-collector ExtensionService
+
+When using Tracing,first,an ExtensionService must be defined for the otel-collector with the cluster-level details for the otel-collector itself.
+For example:
+
+```yaml
+apiVersion: projectcontour.io/v1alpha1
+kind: ExtensionService
+metadata:
+  namespace: projectcontour
+  name: otel-collector
+spec:
+  protocol: h2c
+  services:
+    - name: otel-collector
+      port: 4317
+  timeoutPolicy:
+    response: 50ms
+```
+
+### Contour Configuration
+
+The `tracing` section of the Contour config file will look like:
+
+```yaml
+tracing:
+  # overallSampling defines the sampling rate of trace data.
+  overallSampling: 100
+  # maxPathTagLength defines maximum length of the request path to extract and include in the HttpUrl tag.
+  maxPathTagLength: 256
+  # customTags defines a list of custom tags with unique tag name.
+  # A customTag can only set one of literal, environment and requestHeaderName
+  customTags:
+    # tagName used to populate the tag name
+    - tagName: literal
+      # literal is a static custom tag value.
+      literal: 'this is literal'
+    - tagName: environment
+      # environment indicates that the label value is obtained from the environment variable.
+      environment: HOST
+    - tagName: requestHeaderName
+      # requestHeaderName indicates which request header the label value is obtained from.
+      requestHeaderName: ':path'
+  # extensionService Identifies the extension service defining the openTelemetry collector service.
+  # formatted as <namespace>/<name>.    
+  extensionService: projectcontour/otel-collector
+```
+
+### HCM Configuration
```

Comment:
can move this down to the "Implementation" section

Reply:
ok